crates/flox-rust-sdk/src/models/environment/path_environment.rs

```diff
@@ -421,12 +414,15 @@ impl<S: TransactionState> PathEnvironment<S> {
     async fn transact_with_manifest_contents(
         &mut self,
         manifest_contents: impl AsRef<str>,
-        nix: &NixCommandLine,
-        system: System,
+        flox: &Flox,
     ) -> Result<(), EnvironmentError2> {
+        debug!("transaction: making temporary environment");
         let mut temp_env = self.make_temporary()?;
+        debug!("transaction: updating manifest");
         temp_env.update_manifest(&manifest_contents)?;
-        temp_env.build(nix, &system).await?;
+        debug!("transaction: building environment");
+        temp_env.build(flox).await?;
```

Comment:
The env-link created during install will not be recorded by nix as a valid gcroot!
once the transaction is done, we should still set a gc root such that the environment is not garbage collected by nix.

Granted, we don't perform gc's with nix at the moment.

Reply:
Good catch